rollup/index.ts

```diff
@@ -146,4 +148,20 @@ const main = async () => {
   });
 };
 
+const ensCache = new Map<string, string | null>();
+
+const getEnsName = async (address: string) => {
+  if (ensCache.has(address)) {
+    return ensCache.get(address);
+  }
+  const name = await getDefaultProvider()
+    .lookupAddress(address)
+    .then((name) => {
+      ensCache.set(address, name);
+      return name;
+    })
+    .catch(() => null);
+  return name;
+};
```

Comment:
```suggestion
const ensCache = new Map<string, string>();

const getEnsName = async (address: string) => {
  if (ensCache.has(address)) {
    return ensCache.get(address)!;
  }
  const ens = await getDefaultProvider().lookupAddress(address);
  const name = ens || address;
  ensCache.set(address, name);
  return name;
};
```

Reply:
what about the case when `lookupAddress(address)` errors out?